frontend/ui/menusorter.lua

```diff
@@ -0,0 +1,211 @@
+--[[--
+This module is responsible for constructing the KOReader menu based on a list of
+menu_items and a separate menu order.
+]]
+
+local DataStorage = require("datastorage")
+local util = require("util")
+local DEBUG = require("dbg")
+local _ = require("gettext")
+
+local MenuSorter = {
+    menu_table = {},
+    orphaned_prefix = _("NEW: "),
+    separator = {
+        id = "----------------------------",
+        text = "KOMenu:separator",
+    },
+}
+
+-- thanks to http://stackoverflow.com/a/4991602/2470572
+-- no need to load lfs here
+local function file_exists(name)
+   local f=io.open(name,"r")
+   if f~=nil then io.close(f) return true else return false end
+end
+
+function MenuSorter:readMSSettings(table, config_prefix)
+    if config_prefix then
+        local config_prefix = config_prefix.."_"
+        local menu_order = DataStorage:getSettingsDir().."/"..config_prefix.."menu_order"
+
+        if file_exists(menu_order..".lua") then
+            return require(menu_order) or {}
+        else
+            return {}
+        end
+    else
+        return {}
+    end
+end
+
+function MenuSorter:sort(item_table, order, config_prefix)
```

Comment:
`MenuSorter:mergeAndSort` is probably a better name here. I also suggest moving config_prefix into the first argument.